lib/fictium/rspec/autocomplete/example.rb

```diff
@@ -38,6 +40,23 @@ def extract_method(example, request)
             action = example.action
             action.method = request.method.downcase.to_sym if action.method.blank?
           end
+
+          def filter_header(header)
+            valid_keys = header.keys.select { |name| valid_header?(name) }
+            header.slice(*valid_keys).transform_keys do |key|
+              (key.start_with?('HTTP_') ? key.sub('HTTP_', '') : key).sub('_', '-')
+            end
+          end
+
+          def valid_header?(name)
+            return false if ignored_header_groups.any? { |group| name.downcase.start_with?(group) }
```

Comment:
You can group the conditions to a one return value, do some refactors to make more readable ;)
```
          def valid_header?(name)
            valid_header_group?(name) && valid_header_value?(name)
          end

          def valid_header_group?(name)
            !(ignored_header_groups.any? { |group| name.downcase.start_with?(group) })
          end

          def valid_header_value?(name)
            !Fictium.configuration.ignored_header_values.include?(name.downcase)
          end
```

Reply:
Yes, but remember to use `.none?`are `exclude?` in those cases instead.